spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -378,4 +506,84 @@ private String getStackTraceAsString(Throwable cause) {
 		return stringWriter.getBuffer().toString();
 	}
 
+	/**
+	 * Override this if you want different header names to be used
+	 * in the sent record.
+	 * @return the default Dlt header names
+	 */
+	protected HeaderNames getHeaderNames() {
+		return DeadLetterPublishingHeaderNamesBuilder
+				.original()
+					.offsetHeader(KafkaHeaders.DLT_ORIGINAL_OFFSET)
+					.timestampHeader(KafkaHeaders.DLT_ORIGINAL_TIMESTAMP)
+					.timestampTypeHeader(KafkaHeaders.DLT_ORIGINAL_TIMESTAMP_TYPE)
+					.topicHeader(KafkaHeaders.DLT_ORIGINAL_TOPIC)
+					.partitionHeader(KafkaHeaders.DLT_ORIGINAL_PARTITION)
+				.exception()
+					.keyExceptionFqcn(KafkaHeaders.DLT_KEY_EXCEPTION_FQCN)
+					.exceptionFqcn(KafkaHeaders.DLT_EXCEPTION_FQCN)
+					.keyExceptionMessage(KafkaHeaders.DLT_KEY_EXCEPTION_MESSAGE)
+					.exceptionMessage(KafkaHeaders.DLT_EXCEPTION_MESSAGE)
+					.keyExceptionStacktrace(KafkaHeaders.DLT_KEY_EXCEPTION_STACKTRACE)
+					.exceptionStacktrace(KafkaHeaders.DLT_EXCEPTION_STACKTRACE)
+				.build();
+	}
+
+	/**
+	 * Container class for the headers that will be added to the produced record.
+	 */
+	public static class HeaderNames {
```

Comment:
This should be moved to the builder, or the builder moved here - otherwise we have a class tangle with each referencing each other.

Reply:
Ok. I think it's probably better to leave in the DLPR so I can access it's properties in a simpler way, if that's ok with you.

Reply:
Actually, looking again, the Builder has a reference to the HeaderNames class, but the HeaderNames class doesn't know about the builder.

Reply:
The DLPR knows about the Builder and the Builder knows about DLPR though. I've moved the Builder to the DLPR.